### ISSUE TITLE
[FR] Use new color states

### ISSUE THREAD

**Issue by chemelli74** [nielsfaber/alarmo-card#53]
### Checklist
- [x] I checked for similar existing requests (both open and closed) before posting.
- [x] My request is generic, other users may benefit from it too.

### Proposal

Use the new theme's color states:

```
  state-alarm-armed-color: '#F44336' # Red
  state-alarm-arming-color: '#FF9800' # Orange
  state-alarm-disarmed-color: '#4CAF50' # Green
  state-alarm-pending-color: '#FF9800' # Orange
  state-alarm-triggered-color: '#F44336' # Red
```

### Additional info
[some example of how the new functionality should look like]

**Comment by Eyco1981** [comment]
Isn't the new color scheme in HA
armed: green
arming/pending: orange/yellow
triggered: red
disarmed: blue
?

**Comment by nielsfaber** [comment]
The colors HA used are defined in https://github.com/home-assistant/frontend/blob/dev/src/resources/ha-style.ts:

```
      --rgb-state-alarm-armed-color: var(--rgb-green-color);
      --rgb-state-alarm-arming-color: var(--rgb-orange-color);
      --rgb-state-alarm-disarmed-color: var(--rgb-grey-color);
      --rgb-state-alarm-pending-color: var(--rgb-orange-color);
      --rgb-state-alarm-triggered-color: var(--rgb-red-color);
```

The colors used by this card are:
```
        --alarm-color-disarmed: var(--label-badge-green);
        --alarm-color-pending: var(--label-badge-yellow);
        --alarm-color-triggered: var(--label-badge-red);
        --alarm-color-armed: var(--label-badge-red);
```

Personally I don't think the colors of this card **have** to be equal to those of the HA alarm panel card.
I personally prefer green for disarmed, red for armed state, orange/yellow for the transition states in between (like a traffic light). I don't like grey to be the disarmed color, since I associate this with unavailable state and I find this color to provide bad contrast in my dashboard. 
In summary, I'm happy with the colors as they are.  But I'm open for a discussion about this.

If the tones of the colors have changed and that is the actual request of @chemelli74 , that's another story. 
 I prefer not to use hard-coded color codes, but use the color names defined by HA.
The current mapping is as follows (defined by HA):
```
      --label-badge-red: #DF4C1E;
      --label-badge-blue: #039be5;
      --label-badge-green: #0DA035;
      --label-badge-yellow: #f4b400;
      --label-badge-grey: #9e9e9e;
```

So the color codes are different to your proposed color codes. What is your list of color codes based on, i.e. where are they used in HA?

**Comment by Eyco1981** [comment]
HA itself seems to be a bit inconsistant. I never saw gray as disarmed - as I used the Button card. There it is blue.![Screenshot_20230113-062720_Firefox.jpg](https://user-images.githubusercontent.com/88980079/212244775-9196a6e8-cdfc-46a9-8e31-ff88ef2074cb.jpg)

**Comment by mrgrlscz** [comment]
Hi humans, this is a solution I found, at card level. Just insert this code (uses card-mod component) at the end of alarmo-card:
card_mod:
  style: |
    ha-card > div.header > div.icon > alarmo-state-badge {
    --alarm-color-disarmed: var(--label-badge-blue);
    --alarm-color-pending: var(--label-badge-orange);
    --alarm-color-triggered: var(--label-badge-red);
    --alarm-color-armed: var(--label-badge-green);
    }

**Comment by chemelli74** [comment]
not stale

**Comment by Sam01101101** [comment]
still requires a permanent resolution

**Comment by BenchPressesBooks** [comment]
definitely not stale yet!

**Comment by nielsfaber** [comment]
This should be resolved by updating alarmo-card to v1.5.0.
@chemelli74  Could you please update the card and confirm the colors are OK now?
You can close the issue if this is the case.

**Comment by chemelli74** [comment]
> This should be resolved by updating alarmo-card to v1.5.0. @chemelli74 Could you please update the card and confirm the colors are OK now? You can close the issue if this is the case.

Thx for the update.
Seems all but "disable" state are now in sync.

![image](https://github.com/nielsfaber/alarmo-card/assets/57354320/11fc550a-e1ce-4cf3-99b3-91a5b21aa2a6)

![image](https://github.com/nielsfaber/alarmo-card/assets/57354320/16c27a2a-a5c4-4611-a1b6-09fe1bfe7da1)

**Comment by nielsfaber** [comment]
@chemelli74 HA did not define a color for the disarmed state, see https://github.com/home-assistant/frontend/blob/eabd8b54f410b90c0c032c8422e04fa888fbad9f/src/resources/ha-style.ts#L126-L134
Alarmo-card uses the `--state-inactive-color` instead.
In my theme (default theme) it looks exactly equal to the HA alarm panel card.

**Comment by TekFan** [comment]
Ok, so this works:
(I admit I'm not very fluent in theming so it took me a while to figure it out, sorry 'bout that...)

  state-alarm_control_panel-armed_away-color: var(--red-color)
  state-alarm_control_panel-armed_custom_bypass-color: var(--red-color)
  state-alarm_control_panel-armed_home-color: var(--red-color)
  state-alarm_control_panel-armed_night-color: var(--red-color)
  state-alarm_control_panel-armed_vacation-color: var(--red-color)
  state-alarm_control_panel-arming-color: var(--orange-color)
  state-alarm_control_panel-disarming-color: var(--orange-color)
  state-alarm_control_panel-pending-color: var(--orange-color)
  state-alarm_control_panel-triggered-color: var(--red-color)
  state-alarm_control_panel-disarmed-color: var(--green-color) (note: only in HA panel at the moment)

The only exception is for the disarmed state that stays grey...
I've seen as well that this state is not defined in the .ts but [  state-alarm_control_panel-disarmed-color: var(--green-color)
] works perfectly with HA original control panel, so I guess it does exist but someone forgot to list it in the .ts file.
So I think Alarmo could use it safely instead of [--state-inactive-color] because otherwise, I don't see how we can change the disarmed state with theming without altering all inactive states in HA.
Ot is there another way to change the disarmed color ?

**Comment by TekFan** [comment]
> @chemelli74 HA did not define a color for the disarmed state, see https://github.com/home-assistant/frontend/blob/eabd8b54f410b90c0c032c8422e04fa888fbad9f/src/resources/ha-style.ts#L126-L134
>
Well, [--state-alarm_control_panel-disarmed-color: var(--grey-color);] does exists and works perfectly in HA panel.
It's just an omission in the doc.

> Alarmo-card uses the `--state-inactive-color` instead.
>
imho, I don't think this should be like that.
My view is that the inactive theme color should be used only for inactive devices and in this case Alarmo is still active, it is in a disarmed state but still active in the background for environmental sensors. 
The disarmed status is an active status of the system where doors, windows, etc. can be opened and (in my opinion) should be green like a traffic light where you can go ahead freely.
When the Alarm is armed, it should be red because you can't open anything, you can't go out.
But if Alarmo had a switch to completely shut it down (no background fire or water leak sensors active, etc.) then I would agree there the status would become 'inactive" and be grey as per HA new dogma.

Anyway, It's just my opinion, I don't want to argue with anyone, as long as we have themes, everyone is able to choose what they see fit, but in this particular case we then need a way to change the disarmed color.

So why not use the [state-alarm_control_panel-disarmed-color] ? which btw is correctly defaulted to grey in the latest HA default theme, guaranteeing that it won't break the new dogma.

**Comment by nielsfaber** [comment]
@TekFan I released v1.5.1 which uses the `state-alarm_control_panel-disarmed-color` but also uses fallbacks for cases where the color is not defined. Hope this settles the issue for you.

**Comment by TekFan** [comment]
Awesome !
This is perfect, now everyone will be satisfied with this solution.
Thank you for your fantastic work.

**Comment by sammyke007** [comment]
I've noticed the new colors and would like to return to the old situation. What is the exact working config please? And where do I put it?

**Comment by TekFan** [comment]
Actually, you don't need cardmod, you can just put it in your theme file.
These variables are only influencing the alarmo-card or the original alarm card anyway.
